### PR TITLE
 Adjust retry conditions in config controller and add test

### PR DESCRIPTION
**Pull request by shomron**
Signed-off-by: Oren Shomron <shomron@gmail.com>

### CODE REVIEW

File: pkg/controller/config/config_controller.go [hunk 1]
```diff
@@ -251,13 +251,17 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
...
 		}
+
+		// If we reach here, the watch set hasn't changed since last reconcile, but we
+		// have unfinished wipe/replay business from the last change.
 	} else {
-		// only re-calculate replays if our watch set has changed
+		// The watch set _has_ changed, so recalculate the replay set.
 		r.needsReplay = nil
-		r.wiped = false
+		r.needsWipe = true
```
Comment (maxsmythe) [inline pkg/controller/config/config_controller.go:264]: Yep, was still getting the conditions wrong. Much easier to reason about things with the new names, thanks!

File: pkg/watch/set.go [hunk 2]
```diff
@@ -107,6 +115,12 @@ func (w *Set) RemoveSet(other *Set) {
 }
 
 func (w *Set) Equals(other *Set) bool {
+	if w == nil && other == nil {
```
Comment (maxsmythe) [inline pkg/watch/set.go:118]: Should these be inside the mutex? I don't know if checking for nil pointers is thread-safe

File: pkg/controller/config/config_controller_test.go [hunk 3]
```diff
@@ -469,100 +470,113 @@ func TestConfig_CacheContents(t *testing.T) {
...
-func (f *fakeOpa) keyFor(obj interface{}) (opaKey, error) {
-	o, ok := obj.(runtime.Object)
-	if !ok {
-		return opaKey{}, fmt.Errorf("expected runtime.Object, got: %T", obj)
-	}
-	gvk := o.GetObjectKind().GroupVersionKind()
-	k, err := cache.MetaNamespaceKeyFunc(obj)
-	if err != nil {
-		return opaKey{}, fmt.Errorf("extracting key: %v", err)
+func TestConfig_Retries(t *testing.T) {
```
Comment (maxsmythe) [inline pkg/controller/config/config_controller_test.go:473]: Thank you for thinking through how to test this

File: pkg/watch/set.go (hunk 2 above)
Comment (shomron) [reply pkg/watch/set.go:118]: This is safe - the method gets a copy of the pointer values as locals - the pointers themselves can't be modified by anyone else, just what is pointed at by the pointers.
Regardless, you can't dereference `w` or `other` to get at their mutex if they were nil :)

File: pkg/controller/config/config_controller_test.go (hunk 3 above)
Comment (shomron) [reply pkg/controller/config/config_controller_test.go:473]: Sure thing. I wanted to add a test for the other retry (when wiping fails) - but didn't have enough time to follow through on it.

File: pkg/watch/set.go (hunk 2 above)
Comment (maxsmythe) [reply pkg/watch/set.go:118]: Fair points :p

**Pull request merged**